### PR TITLE
fix(#7670): Ensure objects unsubscribe from staleness when destroyed

### LINKED ISSUE #7670: Staleness indication inconsistency misapplied to layouts based on navigation.
#### Summary
The display gets confused when to show / not show staleness indicator after visiting stale providing object in navigation.

I theorize that pub/sub isn't un-subbing somewhere.

#### Expected vs Current Behavior
If you are on a Display layout where one object/component goes stale, only that component indicates staleness.  What I'd expect.

But, if you visit that object directly on the browse navigation (outside a display layout), there after, the staleness affect not only appears on the object/component but the whole display layout whether the object/component is part of the display layout or not.  If it is part of the layout, the layout is stale and the subcomponent is stale as well.  This seems incorrect. 

#### Steps to Reproduce
1. Create a Sine Wave Generator (Example) and enable Staleness Updates.
2. Create a "Display Layout", you can leave it empty.
3. Click Sine Wave Generator from Navigation - Notice the Staleness indicator occurs occasionally.
4. Visit the empty "Display Layout", notice it indicates staleness occasionally, though Sinewave isn't on the display.
5. To compare, visit the display layout, from the browser navigation, drag the sign-wave onto the blank display.
6. Now you have a display that both the object and the display are showing staleness.
7. Save the layout, reload the page... visit only the display layout, notice only the sinewave goes stale.
8. Visit the sine wave generator that is not part of the display, revisit the display and now both display and component go stale.

#### Environment

* Open MCT Version: 3.2.1 (Feb Build)
* Deployment Type: I think it's prod build.
* OS: Windows 10
* Browser: Latest Chrome 123, Latest Firefox 115.

#### Impact Check List

- [x] Data loss or misrepresented data?
- [ ] Regression? Did this used to work or has it always been broken?
- [ ] Is there a workaround available?
- [x] Does this impact a critical component?
- [ ] Is this just a visual bug with no functional impact?
- [ ] Does this block the execution of e2e tests?
- [ ] Does this have an impact on Performance?

### PR DESCRIPTION
**Pull request by jvigliotta**
Closes #7670

### Describe your changes:

Clearing any existing staleness subscriptions before showing a new object.
Adding a new test to make sure this doesn't regress.
~~Added new functionality to `createDomainObjectWithDefaults` to be able to set toggles when creating test objects.~~
Using new timeAPI methods in SWG Staleness provider.

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing document](https://github.com/nasa/openmct/blob/master/CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/nasa/openmct/pulls) for the same update/change?
* [ ] Is this a [notable change](../docs/src/process/release.md) that will require a special callout in the release notes? For example, will this break compatibility with existing APIs or projects that consume these plugins?

### Author Checklist

* [x] Changes address original issue?
* [x] Tests included and/or updated with changes?
* [x] Has this been smoke tested?
* [x] Have you associated this PR with a `type:` label? Note: this is not necessarily the same as the original issue.
* [ ] Have you associated a milestone with this PR? Note: leave blank if unsure.
* [x] Testing instructions included in associated issue OR is this a dependency/testcase change?

### Reviewer Checklist

* [x] Changes appear to address issue?
* [x] Reviewer has tested changes by following the provided instructions?
* [x] Changes appear not to be breaking changes?
* [x] Appropriate automated tests included?
* [x] Code style and in-line documentation are appropriate?

### CODE REVIEW

File: example/generator/SinewaveStalenessProvider.js [hunk 1]
```diff
@@ -61,10 +59,7 @@ export default class SinewaveLimitProvider extends EventEmitter {
   subscribeToStaleness(domainObject, callback) {
     const id = this.#getObjectKeyString(domainObject);
 
-    if (this.#isRealTime === undefined) {
-      this.#updateRealTime(this.#openmct.time.getMode());
-    }
-
+    this.#realtimeCheck();
```
Comment (ozyx) [inline example/generator/SinewaveStalenessProvider.js:62]: this method name slaps

File: src/ui/components/ObjectView.vue [hunk 2]
```diff
@@ -352,6 +352,10 @@ export default {
     show(object, viewKey, immediatelySelect, currentObjectPath) {
       this.updateStyle();
 
+      if (this.domainObject) {
```
Comment (shefalijoshi) [inline src/ui/components/ObjectView.vue:355]: Instead of this could we add the unsubscribe to `this.triggerStalenessSubscribe`?

**Review by shefalijoshi — CHANGES_REQUESTED** [review]

File: src/ui/components/ObjectView.vue (hunk 2 above)
Comment (jvigliotta) [reply src/ui/components/ObjectView.vue:355]: we can't, we set the domain object a little further down and triggerStalenessSubscribe is called after that, it'd erase the object that was just set.

**Review by shefalijoshi — APPROVED** [review]

File: e2e/appActions.js [hunk 3]
```diff
@@ -101,8 +101,19 @@ async function createDomainObjectWithDefaults(
...
+      // Locate the slider span associated with the checkbox
+      const slider = input.locator('+ .c-toggle-switch__slider');
+      const isChecked = await input.isChecked();
+      if ((value === 'true' && !isChecked) || (value === 'false' && isChecked)) {
+        await slider.click();
+      }
+    } else {
+      await input.fill('');
+      await input.fill(value);
+    }
```
Comment (ozyx) [inline e2e/appActions.js:116]: we ought to move away from this pattern in favor of improving a11y for the different components in a form. i can see this function growing in size and complexity and becoming less and less maintainable as more tests depend on it

Comment (ozyx) [reply e2e/appActions.js:116]: maybe we can sync and find a solution

**Review by ozyx — CHANGES_REQUESTED** [review]

File: e2e/tests/functional/staleness.e2e.spec.js [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+    await page.goto(folder.url);
+
+    // Verify that staleness is not shown
+    const objectView = page.locator(objectViewSelector);
+    const containsStaleClass = await objectView.evaluate((node, staleClass) => {
+      return node.classList.contains(staleClass);
+    }, isStaleClass);
+    await expect(containsStaleClass).toBeFalsy();
+  });
+});
```
Comment (ozyx) [inline e2e/tests/functional/staleness.e2e.spec.js:66]: We should prefer to use assertions to wait for selector states
```suggestion
    // Navigate to the stale object
    await navigateToObjectWithRealTime(page, staleSWG.url);

    // Assert that staleness is shown
    await expect(page.locator(`.c-object-view .${isStaleClass}`)).toBeAttached({
      timeout: 30 * 1000 // Give 30 seconds for the staleness to be updated
    });

    // Immediately navigate to the folder
    await page.goto(folder.url);

    // Verify that staleness is not shown
    await expect(page.locator(`.c-object-view .${isStaleClass}`)).not.toBeAttached();
  });
});
```

**Review by ozyx — APPROVED** [review]
LGTM, one minor nitpick you can take or leave

**Pull request merged**